### PR TITLE
[LSM-clean] simplify struct

### PR DESCRIPTION
**Pull request by feiwang3311**

### CODE REVIEW

File: src/main/scala/lms/thirdparty/c_lib_utils.scala [hunk 1]
```diff
@@ -50,13 +50,13 @@ trait CCodeGenLibStruct extends ExtendedCCodeGen {
   }
 
   override def traverse(n: Node): Unit = n match {
-    case Node(s, "lib-struct", List(Const(m: Manifest[_])), _) =>
-      emit(s"${remap(m)} ")
+    case Node(s, "lib-struct", List(Const(m: String)), _) =>
+      emit(s"$m ")
       shallow(s)
       emitln(";")
```
Comment (Kraks) [inline src/main/scala/lms/thirdparty/c_lib_utils.scala:56]:
```suggestion
      esln"$m $s;"
```
:)

File: src/main/scala/lms/thirdparty/c_lib_utils.scala [hunk 2]
```diff
@@ -27,8 +27,8 @@ trait CCodeGenCMacro extends ExtendedCCodeGen {
 }
 
 trait LibStruct { b : Base =>
-  def newStruct[T: Manifest]: Rep[T] = {
-    Wrap[T](Adapter.g.reflectUnsafe("lib-struct", Backend.Const(manifest[T])))
+  def newStruct[T: Manifest](type_name: String): Rep[T] = {
```
Comment (Kraks) [inline src/main/scala/lms/thirdparty/c_lib_utils.scala:30]:
```suggestion
  def newStruct[T: Manifest](typeName: String): Rep[T] = {
```

https://docs.scala-lang.org/style/naming-conventions.html

**Comment by Kraks** [comment]
Except for those minor suggestions, I'm okay with this change, but I wonder what's the rationale behind it? It seems to be more flexible (and more unsafe) to control the generated type, and we don't need to provide the type mapping in the `remap` function?

**Comment by feiwang3311** [comment]
> Except for those minor suggestions, I'm okay with this change, but I wonder what's the rationale behind it? It seems to be more flexible (and more unsafe) to control the generated type, and we don't need to provide the type mapping in the `remap` function?

The rationale is that if we only need to `declare a C struct` for a library, this allows us to only add code at one place. 
typeMap is needed only if the type is also used independently (such as in a function parameter).

**Review by Kraks — APPROVED** [review]
LGTM

**Pull request merged**